lib/Ix/Validators.pm

```diff
@@ -101,6 +101,7 @@ sub boolean {
   my sub is_domain {
     my $value = shift;
     return unless defined $value and $value =~ /\A$domain_re\z/;
+    return unless length($value) < 256;
```

Comment:
The length of any one domain name is really limited to 253 characters.

This is because:

  * Names are limited to 255 octets
  * A label (63 characters) is really a length byte, followed by that number of characters
  * A domain (253 characters) is a series of labels terminated by a null label (so the final length byte is 0)

So test.com (length: 8) is really 10 bytes:

````
length: 4  t         e        s         t        length: 3   c        o        m        length: 0
00000100   01110100  01100101 01110011  01110100 00000011    01100011 01101111 01101101 00000000
````

Note that test.com and test.com. (trailing period) are equivalent (as far as we are concerned) so if counting length we should always remove the final period if one is provided.